pkg/asset/cluster/tfvars.go

```diff
@@ -359,6 +359,14 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			bootstrapIgnStub = string(shim)
 		}
 
+		var osImage azure.OSImage
+		if installConfig.Config.ControlPlane.Platform.Azure != nil {
+			osImage = installConfig.Config.ControlPlane.Platform.Azure.OSImage
+		}
+		if osImage.Publisher == "" && installConfig.Config.Platform.Azure.DefaultMachinePlatform != nil {
+			osImage = installConfig.Config.Platform.Azure.DefaultMachinePlatform.OSImage
+		}
```

Comment:
nit: the `MasterConfigs` that are already being passed have `.image`, so I think you could just use that instead of passing a separate parameter.

Reply:
Indeed, that is much better. Fixed.